File: ecmascript/difference-of-squares/difference-of-squares.js
function* range(start, end) {
  let i = start;
  while (i <= end)
    yield i++;
}

const sumFunc = (acc, x) => acc + x;

const Squares = (n) => {
  const numbers = [...range(1, n)];
  const sums = numbers.reduce(sumFunc);

  const squareOfSums = sums * sums;
  const sumOfSquares = numbers.map(x => x * x).reduce(sumFunc);
  const difference = squareOfSums - sumOfSquares;

  return { squareOfSums, sumOfSquares, difference };
};

export default Squares;

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  VERSION = 1

  def initialize(n)
    @n = n
  end

  def square_of_sums
    (1..@n).inject(:+)**2
  end

  def sum_of_squares
    (1..@n).inject(0) { |a, e| a + e**2 }
  end

  def difference
    square_of_sums - sum_of_squares
  end
end
